File: java/src/main/java/heritage/Etudiant.java
package heritage;

import java.util.ArrayList;
import java.util.List;

public class Etudiant extends Personne {
    private String numeroEtudiant;
    private double moyenne;
    private List<Cours> listeCours;

    public Etudiant(String nom, int age, String numeroEtudiant, double moyenne) {
        super(nom, age); // obligatoire
        this.numeroEtudiant = numeroEtudiant;
        this.moyenne = moyenne;
        this.listeCours = new ArrayList<>();
    }

    public void ajouterCours(Cours c) {
        listeCours.add(c);
    }

    public List<Cours> getListeCours() {
        return listeCours;
    }

    @Override
    public String toString() {
        return super.toString() +
                " | ID: " + numeroEtudiant +
                " | Moyenne: " + moyenne;
    }
}

File: python/src/etudiant.py
from .personne import Personne
from .cours import Cours


class Etudiant(Personne):
    def __init__(self, nom: str, age: int, numero_etudiant: str, moyenne: float):
        super().__init__(nom, age)  # obligatoire
        self.numero_etudiant = numero_etudiant
        self.moyenne = moyenne
        self.liste_cours = []

    def ajouter_cours(self, cours: Cours):
        self.liste_cours.append(cours)

    def __str__(self):
        cours_str = ", ".join(c.nom_cours for c in self.liste_cours) if self.liste_cours else "Aucun"
        return (
            f"{super().__str__()} | "
            f"ID: {self.numero_etudiant} | "
            f"Moyenne: {self.moyenne} | "
            f"Cours: {cours_str}"
        )

    def __repr__(self):
        return (
            f"Etudiant({self.nom!r}, {self.age!r}, "
            f"{self.numero_etudiant!r}, {self.moyenne!r})"
        )